logger/logger.go

```diff
@@ -0,0 +1,50 @@
+// Copyright 2020 Sonatype Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package logger has functions to obtain a logger, and helpers for setting up where the logger writes
+package logger
+
+import (
+	"os"
+	"path"
+
+	"github.com/sirupsen/logrus"
+)
+
+// DefaultLogFile can be overriden to use a different file name for upstream consumers
+var DefaultLogFile = "nancy.combined.log"
+
+// Logger can be obtained from outside the package
+var Logger *logrus.Logger
+
+func init() {
+	file, err := os.OpenFile(getLogFileLocation(), os.O_CREATE|os.O_APPEND|os.O_WRONLY, 0644)
+	if err != nil {
+		Logger.Fatal(err)
+	}
+
+	logger := &logrus.Logger{
+		Out:          file,
+		Level:        logrus.DebugLevel,
+		Formatter:    &logrus.JSONFormatter{},
```

Comment:
I assume this is going to be more for human consumption.... if im being honest line after line of json output is not super readable to me so I'd vote for switching this out

Reply:
I kinda went with JSON because then you could consume it programmatically easier. Fine with using a TextFormatter, too!

Reply:
up to you really.....in all honesty i cant see a usecase where consuming it programmatically would happen but if you have a usecase sure why not :shipit:

Reply:
I don't, just used to planning for the worst. I'll switch to Text!

Reply:
Ok I switched to Text and I hate it, BECAUSE when you use `WithFields` it doesn't really output the field with it. You know a way around this?

Reply:
What does it give you?? Docs show it should look like this

![](https://camo.githubusercontent.com/b9d0e424bfc6378e79b90de33b983ef5bae2f578/687474703a2f2f692e696d6775722e636f6d2f505937714d77642e706e67)

Reply:
I get effectively:

```
time="2020-03-12T18:04:45-08:00" level=debug msg="Successful in communicating with IQ Server" res="{None http://localhost:8070/ui/links/application/testapp/report/4ca4b65c2eff4e4ca922fa608c78bbd2 false }"
```
What I don't like is to get it to do what you see, I have to take a struct and you know, add a field for each field, rather than be lazy and log the whole thing as "res". It's totally me being lazy, but it also allows someone to add fields to a struct and they get logged by default?

Reply:
ah i see....JsonFormatter it is :) #imAlsoLazy

Reply:
Note this is because it's to file, not to console, but I imagine to console it wouldn't print the object fields.

Reply:
> ....JsonFormatter it is

Agreed! Stick with what was working before I stirred the pot. Thanks for looking into it!